### PR TITLE
restore SDPA in gpt neo after 4.45

### PR DESCRIPTION
**Pull request by eaidova**
# What does this PR do?

bettertransformers is not compatible with gpt-neo starting 4.45, but transformers only provide eager and flash_attn attention implementations, this PR adds sdpa for gpt_neo

### CODE REVIEW

**Review by AlexKoff88 — APPROVED** [review]

**Comment by AlexKoff88** [comment]
I am ok with the changes but I am thinking that next time we can just drop the support of this model as it is the legacy one.

**Comment by eaidova** [comment]
failed tests are not related to changes, it is issue with nonbackward compatible changes in openvino (should be passed in next nightly)

File: optimum/exporters/openvino/model_patcher.py [hunk 1]
```diff
@@ -2654,6 +2654,96 @@ def __exit__(self, exc_type, exc_value, traceback):
...
+        attention_mask=attention_mask,
+        layer_past=layer_past,
+        head_mask=head_mask,
+        use_cache=use_cache,
+        output_attentions=output_attentions,
+        cache_position=cache_position,
+    )
+
+
+# Adopted from https://github.com/huggingface/optimum/blob/main/optimum/bettertransformer/models/attention.py#L185
```
Comment (eaidova) [inline optimum/exporters/openvino/model_patcher.py:2681]:
```suggestion
# Adapted from https://github.com/huggingface/optimum/blob/main/optimum/bettertransformer/models/attention.py#L185
```

**Comment by eaidova** [comment]
diffusers tests fixes are covered by https://github.com/huggingface/optimum-intel/pull/1094

**Comment by AlexKoff88** [comment]
Merging as a hotfix

**Pull request merged**